lib/config/flat-config-schema.js

```diff
@@ -262,6 +263,24 @@ const booleanSchema = {
     validate: "boolean"
 };
 
+/** @type {ObjectPropertySchema} */
+const severitySchema = {
+    merge(first, second) {
+        const value = second === void 0 ? first : second;
+
+        if (typeof value === "boolean") {
+            return value ? "warn" : "off";
+        }
+
+        return normalizeSeverityToString(value);
```

Comment:
Thoughts about normalizing to number values here, for consistency with rule severities?

For example:

```js
// eslint.config.js
module.exports = [{
    linterOptions: {
        reportUnusedDisableDirectives: "error"
    },
    rules: {
        "no-undef": "error"
    }
}];
```

```
eslint --print-config foo.js
```

Currently:

```json
{
  "linterOptions": {
    "reportUnusedDisableDirectives": "error"
  },
  "languageOptions": {
    "ecmaVersion": "latest",
    "sourceType": "module",
    "parser": "espree@9.6.1",
    "parserOptions": {},
    "globals": {}
  },
  "plugins": [
    "@"
  ],
  "rules": {
    "no-undef": [
      2
    ]
  }
}
```

If we normalize to numbers:

```json
{
  "linterOptions": {
    "reportUnusedDisableDirectives": 2
  },
  "languageOptions": {
    "ecmaVersion": "latest",
    "sourceType": "module",
    "parser": "espree@9.6.1",
    "parserOptions": {},
    "globals": {}
  },
  "plugins": [
    "@"
  ],
  "rules": {
    "no-undef": [
      2
    ]
  }
}
```

Reply:
Seems reasonable. I updated to normalize to a number here as it was trivial to do so.

As I commented before, I'm still curious to learn about the long-term plans for strings vs. numbers, if there are any plans around this.

Reply:
There are no long-term plans. Strings have always just been aliases for numbers, and normalizing to numbers allows us to avoid string comparisons.